extmod/modframebuf.c

```diff
@@ -174,11 +174,107 @@ STATIC mp_obj_t framebuf1_text(size_t n_args, const mp_obj_t *args) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(framebuf1_text_obj, 4, 5, framebuf1_text);
 
+// Fill a rectangle.
+STATIC void framebuf1_fillbox(const mp_obj_framebuf1_t *fb, int x, int y, int width, int height, uint8_t color) {
+    if (x < 0 || y < 0 || y >= fb->height || x >= fb->width || height <= 0 || width <= 0) {
+        // No operation needed.
+        return;
+    }
+
+    // clip to the display
+    x = MAX(MIN(x, fb->width), 0);
+    y = MAX(MIN(y, fb->height), 0);
+    int xend = MIN(fb->width, x + width);
+    int yend = MIN(fb->height, y + height);
+    int start_row = y >> 3;
+    int end_row = yend >> 3;
+    uint8_t start_mask, end_mask;
+
+    if (color) {
+        if (end_row == start_row) {
```

Comment:
Yes, it handles the case when the start and end is in the same byte, so the mask has 0's on both sides. The case below can only handle case where start_mask has 0's at the top, and end_mask -- at the bottom.